core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -62,19 +62,22 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
 
   def resizeThreadPool(newSize: Int): Unit = {
     def migratePartitions(newSize: Int): Unit = {
+      val allRemovedPartitionsMap = mutable.Map[TopicPartition, InitialFetchState]()
       fetcherThreadMap.forKeyValue { (id, thread) =>
-        val partitionStates = removeFetcherForPartitions(thread.partitions)
+        val partitionStates = thread.removeAllPartitions()
```

Comment:
Originally, we call `removeFetcherForPartitions`, where we will filter out the `InitialFetchState == null` partitions [here](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/AbstractFetcherThread.scala#L742). I'm not sure if this is something we should care about, but I think we'd better to keep the filter logic to avoid the NPE thrown.

Reply:
I think unless there is a  null as  value in `partitionStates.partitionStateMap`,  there is no NPE thrown. I think `partitionMapLock` make sure there is no other thread changing `partitionStates` in this process. The original version need to filter the null value because `partitionStates.stateValue` might return null when the partition is not included in `partitionStates.partitionStateMap`.

Reply:
Make sense!